### PR TITLE
User Profile translations - value put in correct field after search

### LINKED ISSUE #27838: User Profile translations - value put in wrong field after search
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

For User Profile attributes, the dialog for adding translations will get values mixed up if you do a search first.

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

Translation should be assigned to the language I searched for.

### Actual behavior

Translation is always assigned to the default language.

### How to Reproduce?

1. Enable internationalization
2. Choose English and French as supported languages with English as the default.
3. Go to a User Profile attribute and open the dialog to add translations for the description
4. Search on "F" for French.
5. Now French is the only language shown.
6. Add a translation for French.
7. Save the attribute changes.
8. Go to Realm Settings --> Localization --> Realm overrides
9. See that the translation I added for French is assigned to English.

### Anything else?

_No response_

### LINKED ISSUE #28103: Deleting translations after attribute deletion
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Translations are not getting deleted when attribute is deleted

### Version

main

### Regression

- [ ] The issue is a regression

### Expected behavior

Translations are getting deleted when attribute is deleted

### Actual behavior

Translations are not getting deleted when attribute is deleted

### How to Reproduce?

Delete attribute and observe that the translation in localization/realm override is not deleted.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by agagancarczyk**
Closes https://github.com/keycloak/keycloak/issues/27838 and closes https://github.com/keycloak/keycloak/issues/28103

### CODE REVIEW

File: js/apps/admin-ui/src/realm-settings/user-profile/AttributesTab.tsx [hunk 1]
```diff
@@ -86,9 +86,15 @@ export const AttributesTab = ({ setTableData }: AttributesTabProps) => {
       if (!config?.attributes) return;
 
       const translationsToDelete = config.attributes.find(
-        (attribute) => attribute.name === attributeToDelete,
+        (attribute: UserProfileAttribute) =>
+          attribute.name === attributeToDelete,
```
Comment (jonkoops) [inline js/apps/admin-ui/src/realm-settings/user-profile/AttributesTab.tsx:90]: This type is already inferrable, let's not add any additional type information if it's not needed.
```suggestion
        (attribute) => attribute.name === attributeToDelete,
```

File: js/apps/admin-ui/src/realm-settings/user-profile/AttributesTab.tsx [hunk 2]
```diff
@@ -86,9 +86,15 @@ export const AttributesTab = ({ setTableData }: AttributesTabProps) => {
...
       )?.displayName;
 
+      const formattedTranslationsToDelete = translationsToDelete?.substring(
```
Comment (jonkoops) [inline js/apps/admin-ui/src/realm-settings/user-profile/AttributesTab.tsx:93]: Could you add a comment here with some explanation as to why we need to slice this string like this?

File: js/apps/admin-ui/src/realm-settings/user-profile/AttributesTab.tsx [hunk 3]
```diff
@@ -120,7 +126,8 @@ export const AttributesTab = ({ setTableData }: AttributesTabProps) => {
         );
 
         const updatedAttributes = config.attributes.filter(
-          (attribute) => attribute.name !== attributeToDelete,
+          (attribute: UserProfileAttribute) =>
+            attribute.name !== attributeToDelete,
```
Comment (jonkoops) [inline js/apps/admin-ui/src/realm-settings/user-profile/AttributesTab.tsx:130]: Same here, type can be inferred.
```suggestion
          (attribute) => attribute.name !== attributeToDelete,
```

File: js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx [hunk 4]
```diff
@@ -116,11 +116,32 @@ export const AddTranslationsDialog = ({
   }, [combinedLocales, translationKey, setValue]);
 
   const handleOk = () => {
-    const formData = getValues();
-    onTranslationsAdded(formData);
+    const formData = getValues() as Translations;
```
Comment (jonkoops) [inline js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx:119]: Looks like this compiles the same without a type-cast, so let's remove it.
```suggestion
    const formData = getValues();
```

File: js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx [hunk 5]
```diff
@@ -234,48 +255,53 @@ export const AddTranslationsDialog = ({
...
-                              className="pf-m-sm pf-u-px-sm"
-                              dataLabel={t("supportedLanguage")}
-                            >
-                              <FormGroup fieldId="kc-supportedLanguage">
-                                {localeToDisplayName(
-                                  locale,
-                                  whoAmI.getLocale(),
-                                )}
+                        {filteredLocales.map((locale) => {
+                          const rowIndex = combinedLocales.findIndex(
```
Comment (jonkoops) [inline js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx:259]: Are you sure `rowIndex` is always unique? It's being used as a `key` here, which can lead to issues if not unique in this loop.

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx (hunk 5 above)
Comment (agagancarczyk) [reply js/apps/admin-ui/src/realm-settings/user-profile/attribute/AddTranslationsDialog.tsx:259]: Thanks @jonkoops I have added the `index` provided by the map(), and I'm using it as a fallback to ensure the uniqueness of the `key`.

**Review by jonkoops — APPROVED** [review]

**Pull request merged**